src/java.base/share/classes/jdk/internal/loader/URLClassPath.java

```diff
@@ -208,7 +208,9 @@ public URLClassPath(URL[] urls, AccessControlContext acc) {
 
         this.unopenedUrls = unopenedUrls;
         this.path = path;
-        this.jarHandler = null;
+        // the application class loader uses the built-in protocol handler to avoid protocol
```

Comment:
It should have been a clean backport. Please remove these two imports from this file: 

```java
import java.util.HashSet;
import java.util.Set;
```

Also please mention about testing you have done for this patch !!

Reply:
@offamitkumar 

The imports are necessary and cannot be removed because they are utilized in the code at lines 952 and 964.

Regarding the testing :
I have a sample Java program that prints "Hello, world." After compiling the code, I created two JAR files from the .class file. When I run the Java command with the created JARs in the classpath and the -Djava.protocol.handler.pkgs system property set to a package name,  when I use the -verbose option along , the second JAR in the classpath gets loaded first.